src/main/java/net/dries007/tfc/common/blocks/LogPileBlock.java

```diff
@@ -0,0 +1,127 @@
+        return defaultBlockState().setValue(AXIS, context.getHorizontalDirection().getAxis());
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public ActionResultType use(BlockState state, World world, BlockPos pos, PlayerEntity player, Hand hand, BlockRayTraceResult result)
+    {
+        if (world.isClientSide()) return ActionResultType.SUCCESS;
+        LogPileTileEntity te = Helpers.getTileEntity(world, pos, LogPileTileEntity.class);
+        ItemStack stack = player.getItemInHand(hand);
+        if (te != null)
+        {
+            if (stack.getItem().is(TFCTags.Items.LOG_PILE_LOGS))
+            {
+                if (!player.isShiftKeyDown()) // shift interaction handled in InteractionManager
+                {
+                    if (te.insertLog(stack.copy()))
+                    {
+                        if (!world.isClientSide())
+                        {
+                            Helpers.playSound(world, pos, SoundEvents.WOOD_PLACE);
+                            if (!player.isCreative())
+                                stack.shrink(1);
+                        }
+                        return ActionResultType.sidedSuccess(world.isClientSide());
+                    }
+                }
+            }
+
+            if (!player.isShiftKeyDown() && player instanceof ServerPlayerEntity)
+            {
+                NetworkHooks.openGui((ServerPlayerEntity) player, te, pos);
+                return ActionResultType.SUCCESS;
+            }
+        }
+
+        return ActionResultType.FAIL;
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public BlockState updateShape(BlockState stateIn, Direction facing, BlockState facingState, IWorld worldIn, BlockPos currentPos, BlockPos facingPos)
+    {
+        if (!worldIn.isClientSide() && worldIn instanceof World)
+        {
+            if (facingState.is(BlockTags.FIRE))
+            {
+                BurningLogPileBlock.tryLightLogPile((World) worldIn, currentPos);
+            }
+        }
+        return super.updateShape(stateIn, facing, facingState, worldIn, currentPos, facingPos);
+    }
+
+    @Override
+    public ItemStack getPickBlock(BlockState state, RayTraceResult target, IBlockReader world, BlockPos pos, PlayerEntity player)
+    {
+        LogPileTileEntity te = Helpers.getTileEntity(world, pos, LogPileTileEntity.class);
+        return te != null ? te.getLog().copy() : ItemStack.EMPTY;
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public void onRemove(BlockState state, World world, BlockPos pos, BlockState newState, boolean isMoving)
+    {
+        InventoryTileEntity te = Helpers.getTileEntity(world, pos, InventoryTileEntity.class);
+        if (te != null)
+            te.onBreak();
+        super.onRemove(state, world, pos, newState, isMoving);
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateContainer.Builder<Block, BlockState> builder)
+    {
+        builder.add(AXIS);
```

Comment:
Doesn't matter in this case, but for future reference, this should always be implemented as a super class call:
```suggestion
        super.createBlockStateDefinition(builder.add(AXIS));
```

that way subclasses can pick up superclass properties without having to explicitly declare them